controllers/rhmi/bootstrapReconciler.go

```diff
@@ -623,27 +623,34 @@ func (r *Reconciler) retrieveConsoleURLAndSubdomain(ctx context.Context, serverC
 
 	r.installation.Spec.MasterURL = consoleRouteCR.Status.Ingress[0].Host
 
-	ok, domain, err := customDomain.GetDomain(ctx, serverClient, r.installation)
-	if err != nil && !ok {
-		log.Warning(err.Error())
-		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("customDomain.GetDomain() failure: %w", err)
-	}
-
-	if ok {
-		r.installation.Spec.RoutingSubdomain = domain
-
-		if r.installation.Status.CustomDomain == nil {
-			r.installation.Status.CustomDomain = &integreatlyv1alpha1.CustomDomainStatus{}
-		}
-		r.installation.Status.CustomDomain.Enabled = true
-		if err != nil {
-			r.installation.Status.CustomDomain.Error = err.Error()
-			r.installation.Status.LastError = err.Error()
+	threeScaleOperatorNs := &corev1.Namespace{}
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: common.ThreeScaleOperatorNamespace}, threeScaleOperatorNs)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			ok, domain, err := customDomain.GetDomain(ctx, serverClient, r.installation)
+			if err != nil && !ok {
+				log.Warning(err.Error())
+				return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("customDomain.GetDomain() failure: %w", err)
+			}
+			if ok {
+				r.installation.Spec.RoutingSubdomain = domain
+				if r.installation.Status.CustomDomain == nil {
+					r.installation.Status.CustomDomain = &integreatlyv1alpha1.CustomDomainStatus{}
+				}
+				r.installation.Status.CustomDomain.Enabled = true
+				if err != nil {
+					r.installation.Status.CustomDomain.Error = err.Error()
+					r.installation.Status.LastError = err.Error()
+				}
+			} else {
+				r.installation.Spec.RoutingSubdomain = strings.TrimPrefix(consoleRouteCR.Status.Ingress[0].RouterCanonicalHostname, "router-default.")
+			}
+			return integreatlyv1alpha1.PhaseCompleted, nil
```

Comment:
A code neatness suggestion. Not so important to fix.

```suggestion
			found, domain, err := customDomain.GetDomain(ctx, serverClient, r.installation)
			if err != nil {
				log.Warning(err.Error())
				return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("customDomain.GetDomain() failure: %w", err)
			}
			if found {
				r.installation.Spec.RoutingSubdomain = domain
				if r.installation.Status.CustomDomain == nil {
					r.installation.Status.CustomDomain = &integreatlyv1alpha1.CustomDomainStatus{}
				}
				r.installation.Status.CustomDomain.Enabled = true
				if err != nil {
					r.installation.Status.CustomDomain.Error = err.Error()
					r.installation.Status.LastError = err.Error()
				}
			if !found {
				r.installation.Spec.RoutingSubdomain = strings.TrimPrefix(consoleRouteCR.Status.Ingress[0].RouterCanonicalHostname, "router-default.")
			}
			return integreatlyv1alpha1.PhaseCompleted, nil
```